### DISCUSSION TITLE
Using SecretProvider for imagePullSecrets

### DISCUSSION THREAD

**Discussion by ogaumond-rq** [Azure/secrets-store-csi-driver-provider-azure#707] [Q&A]
Does anyone has a full working sample of using KeyVault for image pull secrets.

There is an example of the SecretProviderClass file to sync a dockerconfigjson secret from KeyVault
https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/master/examples/sync-as-kubernetes-secret/dockerconfigjson_synck8s_v1alpha1_secretproviderclass.yaml

I am able to load this secret in a volume of a given pod. However, I don't understand how I can let the pod know that it should use this secret to pull the image. In fact the SecretProviderClass is created, but I don't see the actual secret ever being created.

**Comment by ogaumond-rq** [comment]
I think I figured what I was doing wrong.
- The actual value of the secret in the KeyVault should be the raw json string for the Docker Config. I was putting it already base64 encoded
- You need to mount the secret in a volume of the pod so that it can then be synchronised to a Kubernetes secret and used as an image pull secret

If I have some time later I may try to contribute a more detailed example that could help the documentation

**Reply by xLEpEX** [reply]
do you were able to have the time :D

**Comment by sathishkumar3009** [comment]
@xLEpEX 

     I'm having the same issue, unable to use the secrets in KV as image pull secret. Could you please help me with this.

**Reply by sathishkumar3009** [reply]
@ogaumond-rq - Have you resolved this issue?
